woocommerce/woocommerce-ios
# HACK Week: Support uploading local files for downloadable products

Part of #5064 


## What

This PR adds an option for uploading document files for downloadable products.

## How
- Updated the bottom sheet for selecting file source with an option for document files on device.
- Added a simple helper file `LocalFileUploader` for handling uploading local files.
- Integrated `UIDocumentPickerViewController` for picking documents on device.
- Updated `MediaAction` and `MediaStore` to support uploading files with local URL.
- Handled uploading documents after picking from device.

📝 Notes:
- There's a [runtime warning](https://github.com/woocommerce/woocommerce-ios/issues/11354) on XCode 15 when uploading files - this will be handled in a separate PR.
- The media library picker is currently displaying only images, so it's not possible to pick documents from there for now.

## Testing instructions

Prerequisite: If you are testing in a simulator, you can add files to your simulator by dragging them to the simulator window.
- Log in to a store and navigate to the Products tab.
- Select any simple product or create one.
- In the product form, select the ellipsis menu (...) > Product Settings > Enable downloadable option.
- Back to the product form, select Downloadable Files > Add File > Document on device.
- Select a document of a supported type, notice that a loading view is displayed, then a screen is displayed for updating the file name.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/a4e97120-b329-47d9-9adc-eb833c4af5d8



---
